### PR TITLE
Maintenance dev

### PR DESCRIPTION
**Pull request by MatHazak**
Maintenance update to make the app available for devices with newer Android APIs.  #40

### CODE REVIEW

File: app/src/main/kotlin/com/github/oryanmat/trellowidget/activity/GeneralPreferenceFragment.kt [hunk 1]
```diff
@@ -32,52 +32,83 @@ class GeneralPreferenceFragment : PreferenceFragment() {
...
             }
-        } else if (key == getString(R.string.pref_display_board_name_key)) {
-            val preference = findPreference(key) as SwitchPreference
-            preference.summary = activity.getString(R.string.pref_display_board_name_desc)
         }
     }
 
-    private fun colorPreference(@StringRes key: Int) = findPreference(getString(key)) as ColorPreference
+    private fun colorPreference(@StringRes key: Int) =
+        findPreference<ColorPreference>(getString(key))!!
```
Comment (oryanm) [inline app/src/main/kotlin/com/github/oryanmat/trellowidget/activity/GeneralPreferenceFragment.kt:122]: Just as an aside, I'd avoid using `!!` too often. Either use `?:` to return another value or throw an more descriptive error, or at least comment why it's safe to assume no null will pop up here.

Comment (MatHazak) [reply app/src/main/kotlin/com/github/oryanmat/trellowidget/activity/GeneralPreferenceFragment.kt:122]: I would agree with you. I just added a try-catch to log the case.

**Comment by johnny5dm** [comment]
@MatHazak just found this app on the Play Store, but shows there as not compatible with my Pixel 6 on Android 14. Did this commit fix that compatibility?

**Comment by MatHazak** [comment]
@johnny5dm 
>Did this commit fix that compatibility?

Actually, the main purpose of this PR was that issue. We need to submit the updated app to Play Store, but we haven't done so yet. More on #40.

**Pull request merged**